clients/js/test/plugins/collection/updateDelegate.test.ts

```diff
@@ -919,3 +933,147 @@ test('it can update an owner-managed plugin on an asset as collection update add
+    asset: assetAddress,
+    newCollection: collectionAddress,
+    newUpdateAuthority: updateAuthority('Collection', [collectionAddress]),
+  }).sendAndConfirm(umi);
+
+  const asset = await fetchAssetV1(umi, assetAddress);
+  t.like(asset, <Partial<AssetV1>>{
+    owner: assetOwner.publicKey,
+    updateAuthority: {
+      type: 'Collection',
+      address: collectionAddress,
+    },
+  });
+});
+
+// Does not work with additionalDelegates
+test('it can add asset to collection as collection additional delegate', async (t) => {
+  const umi = await createUmi();
+  const assetOwner = umi.identity;
+  const collectionUmi = await createUmi();
+
+  const assetSigner = generateSigner(umi);
+  const assetAddress = assetSigner.publicKey;
+  await createV2(umi, {
+    asset: assetSigner,
+    name: 'My Asset',
+    uri: 'https://example.com/my-asset.json',
+  }).sendAndConfirm(umi);
+
+  const collectionSigner = generateSigner(collectionUmi);
+  const collectionAddress = collectionSigner.publicKey;
+  await createCollection(collectionUmi, {
+    collection: collectionSigner,
+    name: 'My Collection',
+    uri: 'https://example.com/my-collection.json',
+  });
+
+  await addCollectionPlugin(collectionUmi, {
+    collection: collectionAddress,
+    plugin: {
+      type: 'UpdateDelegate',
+      // Some other address
+      authority: { type: 'Address', address: generateSigner(umi).publicKey },
+      // Who we want to be able to add collection assets
+      additionalDelegates: [assetOwner.publicKey],
+    },
+  }).sendAndConfirm(collectionUmi);
+
+  await updateV2(umi, {
+    asset: assetAddress,
+    newCollection: collectionAddress,
+    newUpdateAuthority: updateAuthority('Collection', [collectionAddress]),
+  }).sendAndConfirm(umi);
+
+  const asset = await fetchAssetV1(umi, assetAddress);
+  t.like(asset, <Partial<AssetV1>>{
+    owner: assetOwner.publicKey,
+    updateAuthority: {
+      type: 'Collection',
+      address: collectionAddress,
+    },
+  });
+});
+
+// Does not work
+test('it can add asset to collection as collection owner', async (t) => {
+  const umi = await createUmi();
+  const assetOwner = umi.identity;
+  const collectionUmi = await createUmi();
+  const collectionOwner = collectionUmi.identity;
+
+  const assetSigner = generateSigner(umi);
+  await createV2(umi, {
+    asset: assetSigner,
+    name: 'My Asset',
+    uri: 'https://example.com/my-asset.json',
+  }).sendAndConfirm(umi);
+
+  const collectionSigner = generateSigner(collectionUmi);
+  const collectionAddress = collectionSigner.publicKey;
+  await createCollection(collectionUmi, {
+    collection: collectionSigner,
+    name: 'My Collection',
+    uri: 'https://example.com/my-collection.json',
+  });
+
+  await addPlugin(umi, {
+    asset: assetSigner.publicKey,
+    plugin: {
+      type: 'UpdateDelegate',
+      authority: { type: 'Address', address: collectionOwner.publicKey },
+      additionalDelegates: [],
+    },
+  }).sendAndConfirm(umi);
```

Comment:
This is confusing as it says "as collection owner" but I don't think it is using collection authority.  Instead its using update delegate which is set to an address that just happens to be called `collectionOwner` in the test.

Reply:
Ah you're right. I think this is the name being wrong. We're making sure overlapping permissions don't break it. Even if it's a weird case.